### PR TITLE
chore: upgrade dependencies to their latest versions

### PR DESCRIPTION
**Pull request by JKRhb**
This PR updates all dependencies except `readable-stream` and `sinon`. Especially the latter one needs some more special treatment to keep working.

### CODE REVIEW

File: index.ts [hunk 1]
```diff
@@ -15,7 +15,7 @@ import ObserveWriteStream from './lib/observe_write_stream'
 import { parameters, refreshTiming, defaultTiming } from './lib/parameters'
 import { isIPv6 } from 'net'
 import { registerOption, registerFormat, ignoreOption } from './lib/option_converter'
-import { CoapServerOptions, requestListener, CoapRequestParams, ParametersUpdate, AgentOptions, CoapPacket, Option, OptionValue } from './models/models'
+import { type CoapServerOptions, type requestListener, type CoapRequestParams, type ParametersUpdate, type AgentOptions, type CoapPacket, type Option, type OptionValue } from './models/models'
```
Comment (Apollon77) [inline index.ts:18]: Would not "import type {...}" also work and be more clear?

File: test/server.ts [hunk 2]
```diff
@@ -654,7 +654,7 @@ describe('server', function () {
             fastForward(10, 1000)
         })
 
-        it('should reply with a confirmable after an ack', function (done) {
+        it.skip('should reply with a confirmable after an ack', function (done) {
```
Comment (Apollon77) [inline test/server.ts:657]: Why do we skip this test?

File: index.ts (hunk 1 above)
Comment (JKRhb) [reply index.ts]: Thank you :) I wasn't aware of this possibility, should now be fixed.

File: test/server.ts (hunk 2 above)
Comment (JKRhb) [reply test/server.ts]: That was an unintended leftover, should now be removed :)

**Review by Apollon77 — APPROVED** [review]

**Comment by Apollon77** [comment]
Approved, merge wehen you are ready

**Comment by JKRhb** [comment]
> Approved, merge wehen you are ready

Awesome, thank you :)

**Pull request merged**